File: leetcode3718/Solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int missingMultiple(int[] nums, int k) {
        Set<Integer> seen = new HashSet<>();
        for(int num : nums) {
            seen.add(num);
        }

        int ans = k;
        while(seen.contains(ans)) {
            ans += k;
        }
        return ans;
    }

    /**
     * runtime - 0ms
     * public int missingMultiple(int[] nums, int k) {
     *    for (int multiple = k; ; multiple += k) {
     *         boolean found = false;
     *         for (int num : nums) {
     *              if (num == multiple) {
     *                   found = true;
     *                   break;
     *              }
     *         }
     *         if (!found) {
     *              return multiple;
     *         }
     *    }
     * }
     */
}

File: leetcode3718/Solution.py
class Solution:
    '''
    def missingMultiple(self, nums: List[int], k: int) -> int:
        present = [False] * 101

        for num in nums:
            present[num] = True

        multiple = k
        while True:
            if multiple > 100 or not present[multiple]:
                return multiple
            multiple += k
        return multiple
    '''

    def missingMultiple(self, nums:List[int], k:int) -> int:
        seen = set(nums)
        ans = k
        while ans in seen:
            ans += k
        return ans
